### PR TITLE
Allow secure variables (environment) on stages, jobs and environments level (#194).

### PR DESCRIPTION
**Pull request by juhijariwala**

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/assets/javascripts/environment_variables.js [hunk 1]
```diff
@@ -138,8 +137,12 @@ EnvironmentVariables = function() {
         });
     };
 
-    function createWrapperElement(tag_type) {
-        return jQuery(document.createElement(tag_type));
+    function createWrapperElement(tag_type, template_text) {
+        if(tag_type){
+          return jQuery(document.createElement(tag_type)).html(template_text);
+        } else {
+          return jQuery(template_text);
```
Comment (jyotisingh) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/environment_variables.js]: Not sure if I understand this well. Did u really mean to return jQuery(template_text). Template text would normally be some html and not a type or a selector.

File: server/webapp/WEB-INF/rails.new/app/assets/javascripts/jquery.environmentVariableEdit.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+(function ($) {
+  $.resetEnvironmentVariableOn = function (options) {
```
Comment (jyotisingh) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/jquery.environmentVariableEdit.js]: I would recommend against writing global functions such as this one. Could you try encapsulating this in a wrapper class instead.

Comment (jyotisingh) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/jquery.environmentVariableEdit.js]: I do not see any tests for this javascript. Could you write them and update the PR.

Comment (jyotisingh) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/jquery.environmentVariableEdit.js]: Also, we use snake case to name javascript files.

**Comment by jyotisingh** [comment]
This fixes most part of #194. The original ticket also suggested support for secure environment variables at pipeline group level. Currently, there is no concept of environment variables (even plain text ones) at pipeline-group level so I am totally fine not implementing that part of it. @arvindsv - what do you say?

File: server/webapp/WEB-INF/rails.new/app/assets/javascripts/environment_variables.js (hunk 1 above)
Comment (juhijariwala) [inline server/webapp/WEB-INF/rails.new/app/assets/javascripts/environment_variables.js]: JQuery is smart about detecting if the "selector" is actually HTML text and creates an element accordingly. http://jsfiddle.net/r16muadb/1/

**Comment by ketan** [comment]
@jyotisingh - I have addressed your feedback and rebased the commit. Would you mind looking at this again?

**Pull request merged**